workspaces/libnpmpublish/test/publish.js

```diff
@@ -903,3 +903,244 @@ t.test('automatic provenance with incorrect permissions', async t => {
+          tarball: 'http://mock.reg/@npmcli/libnpmpublish-test/-/@npmcli/libnpmpublish-test-1.0.0.tgz',
+        },
+      },
+    },
+    access: 'public',
+    _attachments: {
+      '@npmcli/libnpmpublish-test-1.0.0.tgz': {
+        content_type: 'application/octet-stream',
+        data: tarData.toString('base64'),
+        length: tarData.length,
+      },
+      '@npmcli/libnpmpublish-test-1.0.0.sigstore': {
+        // Can't match data against static value as signature is always
+        // different.
+        // Can't match length because in github actions certain environment
+        // variables are present that are not present when running locally,
+        // changing the payload size.
+        content_type: 'application/vnd.dev.sigstore.bundle+json;version=0.1',
+      },
+    },
+  }
+
+  /*
+  const oidcSrv = MockRegistry.tnock(t, oidcURL)
+  oidcSrv.get('/?audience=sigstore', undefined, {
+    authorization: `Bearer ${requestToken}`,
+  }).reply(200, { value: idToken })
+  */
+
+  const fulcioSrv = MockRegistry.tnock(t, fulcioURL)
+  fulcioSrv.matchHeader('Content-Type', 'application/json')
+    .post('/api/v2/signingCert', {
+      credentials: { oidcIdentityToken: idToken },
+      publicKeyRequest: {
+        publicKey: {
+          algorithm: 'ECDSA',
+          content: /.+/i,
+        },
+        proofOfPossession: /.+/i,
+      },
+    })
+    .reply(200, certificateResponse)
+
+  const rekorSrv = MockRegistry.tnock(t, rekorURL)
+  rekorSrv
+    .matchHeader('Accept', 'application/json')
+    .matchHeader('Content-Type', 'application/json')
+    .post('/api/v1/log/entries')
+    .reply(201, rekorEntry)
+
+  registry.getVisibility({ spec, visibility: { public: true } })
+  registry.nock.put(`/${spec.escapedName}`, body => {
+    const bundleAttachment = body._attachments['@npmcli/libnpmpublish-test-1.0.0.sigstore']
+    const bundle = JSON.parse(bundleAttachment.data)
+    const provenance = JSON.parse(Buffer.from(bundle.dsseEnvelope.payload, 'base64').toString())
+
+    t.hasStrict(body, packument, 'posted packument matches expectations')
+    t.hasStrict(provenance.subject[0],
+      expectedSubject,
+      'provenance subject matches expectations')
+    t.hasStrict(provenance.predicate.buildType,
+      'https://github.com/npm/cli/gitlab/v0alpha1',
+      'buildType matches expectations')
+    t.hasStrict(provenance.predicate.builder.id,
+      `${serverUrl}/${repository}/-/runners/${runnerID}`,
+      'builder id matches expectations')
+    t.hasStrict(provenance.predicate.invocation.configSource,
+      expectedConfigSource,
+      'configSource matches expectations')
+    return true
+  }).reply(201, {})
+
+  const ret = await publish(manifest, tarData, {
+    ...opts,
+    provenance: true,
+    fulcioURL: fulcioURL,
+    rekorURL: rekorURL,
+  })
+  t.ok(ret, 'publish succeeded')
+  t.match(log, [
+    ['notice', 'publish',
+      'Signed provenance statement with source and build information from GitHub Actions'],
```

Comment:
Oh I think we need to update the log were we publish to handle gitlab as well:
```suggestion
      'Signed provenance statement with source and build information from GitLab CI'],
```

Reply:
https://github.com/npm/cli/blob/baae2d60109c3362c38ad09c231a32c84c26f63a/workspaces/libnpmpublish/lib/publish.js#L207

Reply:
Done.